internal/otel/config/config.go

```diff
@@ -150,6 +149,13 @@ func buildComponent(id component.ID, p *Params) (any, error) {
 		}
 		return extensions.OauthClientCfg(p.ClientID, p.ClientSecret), nil
 	default:
+		if id == p.ExporterConfig.ID {
```

Comment:
Can we make a type for these `p.ExporterConfig.ID` to avoid this nested if case?

Reply:
Since we don't have an interface anymore we can refactor this to a switch statement. I ran into issues in the exporter config and needed to refactor it further as well but we can get to that later

Reply:
Hmm ok after trying to do this, I understand the problem with the uniqueness of components and that struct `component.ID` 😓 I feel like the OTEL library should have abstracted this better, but I guess we can do that ourselves... Ignore this comment for now, I can make a PR on top with something!

Reply:
I mistyped last night. I ran into issues on the otelgrpc exporter because the component.ID is reused on the exporter which means this switch statement won't work anymore

Reply:
Because I had to separate exporters out to a separate set of functions I'm going to leave this as is for now and we can get back to it on the next PR?